### ISSUE TITLE
Error executing 'static:sockets.validateSession' in plugin 'nodebb-plugin-2factor'

### ISSUE THREAD

**Issue by boydaihungst** [julianlam/nodebb-plugin-2factor#67]
After add 2FA for admin account. I got this error every time login.
This is repo: https://github.com/boydaihungst/nodebb-error
 Using this command to start `docker-compose up --build`
```
mio-translator-forum_1  | 2021-11-19T04:58:07.405Z [4567/70] - error: [plugins] Error executing 'static:sockets.validateSession' in plugin 'nodebb-plugin-2factor'
mio-translator-forum_1  | Error: [[2factor:second-factor-required]]
mio-translator-forum_1  |     at plugin.checkSocket (/usr/src/app/node_modules/nodebb-plugin-2factor/library.js:210:9)
mio-translator-forum_1  |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
mio-translator-forum_1  |     at async Object.fireStaticHook [as static] (/usr/src/app/src/plugins/hooks.js:209:5)
mio-translator-forum_1  |     at async Hooks.fire (/usr/src/app/src/plugins/hooks.js:105:17)
mio-translator-forum_1  |     at async validateSession (/usr/src/app/src/socket.io/index.js:210:17)
mio-translator-forum_1  |     at async onConnect (/usr/src/app/src/socket.io/index.js:88:3)
mio-translator-forum_1  | 2021-11-19T04:58:07.408Z [4567/70] - error: uncaughtException: [[2factor:second-factor-required]]
mio-translator-forum_1  | Error: [[2factor:second-factor-required]]
mio-translator-forum_1  |     at plugin.checkSocket (/usr/src/app/node_modules/nodebb-plugin-2factor/library.js:210:9)
mio-translator-forum_1  |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
mio-translator-forum_1  |     at async Object.fireStaticHook [as static] (/usr/src/app/src/plugins/hooks.js:209:5)
mio-translator-forum_1  |     at async Hooks.fire (/usr/src/app/src/plugins/hooks.js:105:17)
mio-translator-forum_1  |     at async validateSession (/usr/src/app/src/socket.io/index.js:210:17)
mio-translator-forum_1  |     at async onConnect (/usr/src/app/src/socket.io/index.js:88:3) {"date":"Fri Nov 19 2021 04:58:07 GMT+0000 (Coordinated Universal Time)","error":{},"exception":true,"os":{"loadavg":[1.1,0.71,0.74],"uptime":673444.95},"process":{"argv":["/usr/local/bin/node","/usr/src/app/app.js"],"cwd":"/usr/src/app","execPath":"/usr/local/bin/node","gid":1000,"memoryUsage":{"arrayBuffers":18370131,"external":22945963,"heapTotal":120328192,"heapUsed":75774488,"rss":175878144},"pid":70,"uid":1000,"version":"v16.13.0"},"stack":"Error: [[2factor:second-factor-required]]\n    at plugin.checkSocket (/usr/src/app/node_modules/nodebb-plugin-2factor/library.js:210:9)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.fireStaticHook [as static] (/usr/src/app/src/plugins/hooks.js:209:5)\n    at async Hooks.fire (/usr/src/app/src/plugins/hooks.js:105:17)\n    at async validateSession (/usr/src/app/src/socket.io/index.js:210:17)\n    at async onConnect (/usr/src/app/src/socket.io/index.js:88:3)","trace":[{"column":9,"file":"/usr/src/app/node_modules/nodebb-plugin-2factor/library.js","function":"plugin.checkSocket","line":210,"method":"checkSocket","native":false},{"column":5,"file":"node:internal/process/task_queues","function":"processTicksAndRejections","line":96,"method":null,"native":false},{"column":5,"file":"/usr/src/app/src/plugins/hooks.js","function":"async Object.fireStaticHook [as static]","line":209,"method":"fireStaticHook [as static]","native":false},{"column":17,"file":"/usr/src/app/src/plugins/hooks.js","function":"async Hooks.fire","line":105,"method":"fire","native":false},{"column":17,"file":"/usr/src/app/src/socket.io/index.js","function":"async validateSession","line":210,"method":null,"native":false},{"column":3,"file":"/usr/src/app/src/socket.io/index.js","function":"async onConnect","line":88,"method":null,"native":false}]}
```
![image](https://user-images.githubusercontent.com/38396158/142567673-474b02cc-0de5-4235-938a-8cfb70c3c3e7.png)

**Comment by julianlam** [comment]
Ah this doesn't seem to be an issue, doesn't crash NodeBB, so it is handled. Can ignore.

**Comment by julianlam** [comment]
Specifically, it is fixed here: nodebb/nodebb@fa01801f645ea980f34a73eed5fb6596c80d1e57

... and only occurs in recent versions of Node.js, where unhandled exceptions cause the process to crash.
